qemu/target-i386/unicorn.c

```diff
@@ -573,6 +624,38 @@ int x86_reg_write(struct uc_struct *uc, unsigned int regid, const void *value)
 {
     CPUState *mycpu = first_cpu;
 
+    switch(regid) {
+        default:
+            break;
+        case UC_X86_REG_FP0 ... UC_X86_REG_FP7:
+            {
+                uint64_t mant = *(uint64_t*) value;
+                uint16_t upper = *(uint16_t*) (value + sizeof(uint64_t));
+                X86_CPU(uc, mycpu)->env.fpregs[regid - UC_X86_REG_FP0].d = cpu_set_fp80(mant, upper);
+            }
+            break;
+        case UC_X86_REG_FPSW:
+            {
+                uint16_t fpus = *(uint16_t*) value;
+                X86_CPU(uc, mycpu)->env.fpus = fpus & ~0x3800;
+                X86_CPU(uc, mycpu)->env.fpstt = (fpus >> 11) & 0x7;
+            }
+            break;
+        case UC_X86_REG_FPCW:
+            *(uint16_t*) value = X86_CPU(uc, mycpu)->env.fpuc;
```

Comment:
this is another bug here: you are reading, but not writing? this can be detected with a testcase/sample, too.